### ISSUE TITLE
Specific terms fine tuning in french

### ISSUE THREAD

**Issue by sarwg** [kyutai-labs/delayed-streams-modeling#25]
First thanks for this amazing work.
I want to fine-tune the en_fr model on specific data like pharma or health data. How can I do that? And how many hours do I need to have some great results.
Thx again for this project.

**Comment by vvolhejn** [comment]
Hi, please see #4 for a discussion about fine-tuning in general. If there are not too many terms, you can try prompting, which is an experimental feature: https://github.com/kyutai-labs/delayed-streams-modeling/blob/96ff217437e8fbb01b4f067e905c234940c0f22b/scripts/transcribe_from_file_via_pytorch_with_prompt.py

You can provide audio and a corresponding transcript that determines how that audio should be transcribed. It's a bit brittle - sometimes the model will refuse to transcribe anything, in particular if it thinks your provided transcript is very unlikely. But if there's not too many terms, we think this can work.

**Comment by jploski** [comment]
> Hi, please see [#4](https://github.com/kyutai-labs/delayed-streams-modeling/issues/4) for a discussion about fine-tuning in general. If there are not too many terms, you can try prompting, which is an experimental feature: https://github.com/kyutai-labs/delayed-streams-modeling/blob/96ff217437e8fbb01b4f067e905c234940c0f22b/scripts/transcribe_from_file_via_pytorch_with_prompt.py
> 
> You can provide audio and a corresponding transcript that determines how that audio should be transcribed. It's a bit brittle - sometimes the model will refuse to transcribe anything, in particular if it thinks your provided transcript is very unlikely. But if there's not too many terms, we think this can work.

I believe this "refuse to transcribe anything" issue is more of a problem in the implementation of the example script you provided. The issue happens when the probability/logit of the prompt tokens you want to force is smaller than that of the padding tokens. In that case the padding tokens, which are also left unmasked, will be predicted instead of the prompt - I'm referring to this snippet:

```
        mask = torch.zeros_like(logits, dtype=torch.bool)
        for t in self.padding_tokens:
            mask[..., t] = True
        mask[..., self.prefix_enforce[0]] = True

        logits[:] = torch.where(mask, logits, float("-inf"))
```
This can be side-stepped / hacked away by only setting mask[..., t] = True for the padding token if the probability/logit is above some threshold (e.g. if logits[..., t].max() > 3.0: mask[..., t] = True).

A bigger problem with the prompting technique is that the model will not always follow the prompt exactly when the input sequence occurs later in audio input - and you have no way to predict how/why it will transcribe it differently. In your own audio example (stt_from_file_with_prompt_pytorch.py --file ../audio/bria.mp3 --prompt_file ../audio/loona.mp3 --prompt_text ...) I can make it transcribe "Loonah" for the prompt, but it will still transcribe it as  "Loona" for later occurrences. While if I force "Loonoh" or "Loonuh", it works.
